src/Settings/Default.hs

```diff
@@ -215,6 +215,44 @@ defaultBuilderArgs = mconcat
     , makeBuilderArgs
     , tarBuilderArgs ]
 
+-- | Disable some warnings in packages we use
+-- | https://github.com/ghc/ghc/blob/master/mk/warnings.mk#L46
+disableWarningArgsStage0 :: Args
+disableWarningArgsStage0 = stage Stage0 ? builder Ghc ? mconcat
+    [ package transformers ? mconcat [ arg "-fno-warn-unused-matches", arg "-fno-warn-unused-imports" ]
+    , package terminfo ? mconcat [ arg "-fno-warn-unused-imports" ] ]
+
+-- XXX: How to express Stage1 and upper?
+-- XXX: What about the commented-out packages?
+disableWarningArgsStage1 :: Args
+disableWarningArgsStage1 = stage Stage1 ? builder Ghc ? mconcat
+    [ package bytestring ? mconcat [ arg "-Wno-inline-rule-shadowing" ]
+    , package haddock ? mconcat [ arg "-Wno-unused-imports", arg "-Wno-deprecations" ]
+--    , package vector ? mconcat [ arg "-Wno-unused-imports", arg "-Wwarn" ]
+    , package directory ? mconcat [ arg "-Wno-unused-imports" ]
+    , package binary ? mconcat [ arg "-Wno-deprecations" ]
+    , package haskeline ? mconcat [ arg "-Wno-deprecations", arg "-Wno-unused-imports", arg "-Wno-redundant-constraints", arg "-Wno-simplifiable-class-constraints" ]
+    , package pretty ? mconcat [ arg "-Wno-unused-imports" ]
+    , package primitive ? mconcat [ arg "-Wno-unused-imports", arg "-Wno-deprecations" ]
+--    , package random ? mconcat [ arg "-Wno-redundant-constraints" ]
+    , package terminfo ? mconcat [ arg "-Wno-unused-imports" ]
+    , package xhtml ? mconcat [ arg "-Wno-unused-imports", arg "-Wno-tabs" ]
+--    , package dphBase ? mconcat [ arg "-Wwarn" ]
+--    , package dphPrimInterface ? mconcat [ arg "-Wwarn"]
+--    , package dphPrimSeq ? mconcat [ arg "-Wwarn"]
+--    , package dphPrimPar ? mconcat [ arg "-Wwarn"]
+--    , package dphLiftedCommon ? mconcat [ arg "-Wwarn"]
+    , package transformers ? mconcat [ arg "-Wno-unused-matches", arg "-Wno-unused-imports", arg "-Wno-redundant-constraints", arg "-Wno-orphans" ]
+    , package base ? mconcat [ arg "-Wno-trustworthy-safe" ]
+    , package ghcPrim ? mconcat [ arg "-Wno-trustworthy-safe" ]
+    , package win32 ? mconcat [ arg "-Wno-trustworthy-safe" ] ]
+
+-- TODO: Disable warnings for the following
+-- GhcLibExtraHcOpts += -Wno-deprecated-flags
```

Comment:
I usually handle these as follows:
```haskell
    pkg <- getPackage
    mconcat [ isLibrary pkg ? ... ]
```
But I could create a combinator to make this terser, e.g. `libraryPackage ?`. Would you like one?